### DISCUSSION TITLE
Registry metadata size.

### DISCUSSION THREAD

**Discussion by hyrious** [npm/feedback#537] [CLI] [answered]
The registry service returns full metadata including all versions for every package. It can become very big in popular libraries. I'm wondering if we can have some way to reduce the size. At least when we do `npm install pkg`, can we just return a small piece of latest versions?

Ref: https://github.com/evanw/esbuild/issues/789#issuecomment-901467782

Some data:
| package name | version count | transferred (gzipped data) | resources (real size) |
| - | - | - | - |
| clean-stack | 15 | 9.4 kB | 30.6 kB |
| esbuild | 274 | 198 kB | 550 kB |
| antd | 674 | 412 kB | 4.7 MB |

Additional info:
Some other api can be considered like https://data.jsdelivr.com/v1/package/npm/esbuild

**Comment by MylesBorins** [comment, marked as answer]
@hyrious this is definitely something that we've been aware of and it definitely doesn't scale. The biggest issue is that dropping certain details could be a a breaking change for various customers. We have done a couple different explorations about data we can drop but we have not been successful at finding data we can drop
